controller/workitem.go

```diff
@@ -111,18 +136,44 @@ func (c *WorkitemController) Update(ctx *app.UpdateWorkitemContext) error {
 	if !authorized {
 		return jsonapi.JSONErrorResponse(ctx, errors.NewForbiddenError("user is not authorized to access the space"))
 	}
+
+	if ctx.Payload.Data.Relationships != nil && ctx.Payload.Data.Relationships.BaseType != nil &&
+		ctx.Payload.Data.Relationships.BaseType.Data.ID != wi.Type {
+		// Store new values of type and version
+		newTypeID := ctx.Payload.Data.Relationships.BaseType
+		newVersion := ctx.Payload.Data.Attributes[workitem.SystemVersion]
+
+		// Remove version and base type from payload
+		delete(ctx.Payload.Data.Attributes, workitem.SystemVersion)
+		ctx.Payload.Data.Relationships.BaseType = nil
+
+		// Ensure we do not have any other change in payload except type change
+		if (app.WorkItemRelationships{}) != *ctx.Payload.Data.Relationships || len(ctx.Payload.Data.Attributes) > 0 {
+			return jsonapi.JSONErrorResponse(ctx, errors.NewBadParameterErrorFromString("cannot update type along with other fields"))
+		}
+
+		// Restore the original values
+		ctx.Payload.Data.Relationships.BaseType = newTypeID
+		ctx.Payload.Data.Attributes[workitem.SystemVersion] = newVersion
+
+		authorized, err := c.authorizeWorkitemTypeEditor(ctx, c.db, wi.SpaceID, creator.(string), currentUserIdentityID.String())
```

Comment:
@jarifibrahim The first thing that should be checked is if the the user is authorized to change WIT. Only then we should proceed with other checks in the request payload.


This piece of code (checking authorization) should be before this line (https://github.com/fabric8-services/fabric8-wit/pull/2202/files#diff-02845b093c442bf1ae790a6ebb6c8da8R142)

Reply:
Agreed. Please see https://github.com/fabric8-services/fabric8-wit/pull/2202/commits/f700956cb88270706614da87382cdb70918e8c17